pkg/cloudprovider/providers/azure/azure_storage.go

```diff
@@ -19,23 +19,49 @@ package azure
 import (
 	"fmt"
 
+	"github.com/Azure/azure-sdk-for-go/arm/storage"
+	"github.com/Azure/go-autorest/autorest/to"
 	"github.com/golang/glog"
 )
 
+const (
+	defaultStorageAccountType  = string(storage.StandardLRS)
+	fileShareAccountNamePrefix = "f"
+)
+
 // CreateFileShare creates a file share, using a matching storage account
 func (az *Cloud) CreateFileShare(name, storageAccount, storageType, location string, requestGB int) (string, string, error) {
 	var err error
 	accounts := []accountWithLocation{}
-	if len(storageAccount) > 0 {
-		accounts = append(accounts, accountWithLocation{Name: storageAccount})
-	} else {
-		// find a storage account
-		accounts, err = az.getStorageAccounts()
-		if err != nil {
-			// TODO: create a storage account and container
-			return "", "", err
+	if len(storageAccount) == 0 {
+		// find a storage account that matches storageType
+		accounts, err = az.getStorageAccounts(storageType)
+		if err != nil || len(accounts) == 0 {
+			storageAccount = getStorageAccountName(fileShareAccountNamePrefix)
+			if location == "" {
+				location = az.controllerCommon.location
+			}
+			if storageType == "" {
+				storageType = defaultStorageAccountType
+			}
+
+			glog.V(2).Infof("azureFile - no storage account found, begin to create a new storage account %s in resource group %s, location: %s, storageType: %s",
+				storageAccount, az.controllerCommon.resourceGroup, location, storageType)
+			cp := storage.AccountCreateParameters{
+				Sku:      &storage.Sku{Name: storage.SkuName(storageType)},
+				Tags:     &map[string]*string{"created-by": to.StringPtr("azure-file")},
```

Comment:
`created-by` value could better be 'azure-k8s-cloud-provider` or so, or we should not set it at all as we do not do it for loadbalancers.

Reply:
it's similiar to 
https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/azure/azure_blobDiskController.go#L526, I would suggest keep it, anyway it's harmless
```
		cp := storage.AccountCreateParameters{
			Sku:      &storage.Sku{Name: storageAccountType},
			Tags:     &map[string]*string{"created-by": to.StringPtr("azure-dd")},
			Location: &location}
		cancel := make(chan struct{})
```